src/metabase/query_processor/dashboard.clj

```diff
@@ -37,6 +37,28 @@
                     :card_id          card-id
                     :dashboardcard_id dashcard-id)))))
 
+(defn- check-request-param [{param-id :id, :as request-param} matching-param]
+  ;; if `request-param` specifies type, then validate that the type is allowed
+  (when (:type request-param)
+    (qp.card/check-allowed-parameter-value-type
+     param-id
+     (or (when (and (= (:type matching-param) :dimension)
+                    (not= (:widget-type matching-param) :none))
+           (:widget-type matching-param))
+         (:type matching-param))
+     (:type request-param))))
+
+(defn- convert-request-param-value [request-param matching-param]
```

Comment:
Can you find a name that describes the intention? (I'm sure it won't be easy.) Also, can you please write a docstring that explains how the result of the function depends on the inputs?

Reply:
This is legacy code that I've simply extracted. And I've tried to investigate why it was done this way, but unfortunately failed to reproduce the issue. It's possible this modification was made to address a specific bug.